js/content/common.js

```diff
@@ -196,16 +196,45 @@ let ExtensionLayer = (function() {
 
     // NOTE: use cautiously!
     // Run script in the context of the current tab
-    self.runInPageContext = function(fun){
-        let script  = document.createElement('script');
-        script.textContent = '(' + fun + ')();';
+    self.runInPageContext = function(fun) {
+        let script  = document.createElement("script");
+        script.textContent = '(' + fun + ")();";
         document.documentElement.appendChild(script);
         script.parentNode.removeChild(script);
     };
 
     return self;
 })();
 
+class Messenger {
+    static sendMessage(msgID, info) {
+        window.postMessage({
+            type: "es_" + msgID,
+            information: info
+        }, '*');
+    }
+
+    static addMessageListener(msgID, fn, autoRemove) {
+        let callback = function(e) {
+            if (e.source !== window) { return; }
+            if (!e.data.type) { return; }
+            if (e.data.type === "es_" + msgID) {
+                fn(e.data.information);
+                if (autoRemove) {
+                    window.removeEventListener("message", callback);
+                }
+            }
+        }
+        window.addEventListener("message", callback);
+    }
+}
+
+// Inject the Messenger class into the DOM, providing the same interface for the page context side
+(function() {
+    let script = document.createElement("script");
+    script.textContent = Messenger;
```

Comment:
This is an implicit `.toString()`, can we make it explicit?

Reply:
Yes